TOOLS/lua/autoload.lua

```diff
@@ -194,19 +204,14 @@ function find_and_add_entries()
     for direction = -1, 1, 2 do -- 2 iterations, with direction = -1 and +1
         for i = 1, MAXENTRIES do
             local file = files[current + i * direction]
-            local pl_e = pl[pl_current + i * direction]
             if file == nil or file[1] == "." then
                 break
             end
 
             local filepath = dir .. file
-            if pl_e then
-                -- If there's a playlist entry, and it's the same file, stop.
-                msg.trace(pl_e.filename.." == "..filepath.." ?")
-                if pl_e.filename == filepath then
-                    break
-                end
-            end
+            local filenames = get_playlist_filenames()
```

Comment:
Should this be called earlier? It appears that you're getting the list of filenames each time this loop runs.

Reply:
I didn't notice this and should avoid it. Thanks for reminding.